PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/action/ProfileFlightRecordAction.java

```diff
@@ -41,42 +40,44 @@
 
 import static com.microsoft.intellij.ui.messages.AzureBundle.message;
 
+// todo: Migrate profile client implementation to track2 sdk
 @Name("Profile Flight Recorder")
 public class ProfileFlightRecordAction extends NodeActionListener {
     private static final Logger logger = Logger.getLogger(ProfileFlightRecordAction.class.getName());
     private static final int ONE_SECOND = 1000;
     private static final int TWO_SECONDS = 2000;
     private final Project project;
+    private final String webAppId;
+
     private WebAppBase appService;
 
     public ProfileFlightRecordAction(WebAppNode webAppNode) {
         super();
         this.project = (Project) webAppNode.getProject();
-        appService = webAppNode.getWebapp();
+        this.webAppId = webAppNode.getWebAppId();
     }
 
     @Override
     protected void actionPerformed(NodeActionEvent nodeActionEvent) {
-        // prerequisite check
-        if (appService.operatingSystem() == OperatingSystem.LINUX &&
-                StringUtils.containsIgnoreCase(appService.linuxFxVersion(), "DOCKER|")) {
-            final String message = message("webapp.flightRecord.error.notSupport.message", appService.name());
-            notifyUserWithErrorMessage(message("webapp.flightRecord.error.notSupport.title"), message);
-            return;
-        }
-        EventUtil.executeWithLog(appService instanceof WebApp ? TelemetryConstants.WEBAPP : TelemetryConstants.FUNCTION,
-                                 "start-flight-recorder", op -> {
-                final IAzureOperationTitle title = AzureOperationBundle.title("appservice|flight_recorder.profile");
-                final AzureTask task = new AzureTask(project, title, true, this::doProfileFlightRecorderAll, AzureTask.Modality.ANY);
-                AzureTaskManager.getInstance().runInBackground(task);
-            });
+        EventUtil.executeWithLog(TelemetryConstants.WEBAPP, "start-flight-recorder", op -> {
+            final IAzureOperationTitle title = AzureOperationBundle.title("appservice|flight_recorder.profile");
+            final AzureTask task = new AzureTask(project, title, true, this::doProfileFlightRecorderAll, AzureTask.Modality.ANY);
+            AzureTaskManager.getInstance().runInBackground(task);
+        });
     }
 
     private void doProfileFlightRecorderAll() {
         try {
+            // prerequisite check
             final String subscriptionId = AzureMvpModel.getSegment(appService.id(), "subscriptions");
             // Always get latest app service status, workaround for https://dev.azure.com/mseng/VSJava/_workitems/edit/1797916
             appService = AzureWebAppMvpModel.getInstance().getWebAppById(subscriptionId, appService.id());
+            if (appService.operatingSystem() == OperatingSystem.LINUX &&
```

Comment:
shall there be a isLinux and isWindows isDocker in appService?

Reply:
For profile, we are now still use the track 1 client as profile is based on track 1 kudu client.

By the way, we could use `appService.getRuntime().getOs()`, do we need to provide a method like `isWindows` or `isDocker`?